src/GraphPanel.java

```diff
@@ -3,20 +3,19 @@
 import java.awt.event.*;
 import java.awt.geom.GeneralPath;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public class GraphPanel extends JPanel {
-    private double zoom = 1.0;
+    private final List<ParametricFunction> polynomials = new ArrayList<>();
     private Vector2D offset = new Vector2D(0,0);
-    private double scale = 50.0;
-    private final Color backgroundColor = new Color(0.13f, 0.16f, 0.2f);
     private Vector2D lastMousePosition = null;
+    private double zoom = 1.0;
+    private double scale = 50.0;
     private JTextField functionField;
-    public JButton resetButton;
-    private ParametricFunction polynomial;
-    private List<Double> zeroPoints;
-    private List<Double> extremePoints;
+    private final Color backgroundColor = new Color(0.13f, 0.16f, 0.2f);
+    List<Color> colours = Arrays.asList(Color.WHITE, Color.BLUE, Color.GREEN, Color.RED, Color.PINK);
```

Comment:
since the class is called "Color", this should be mirrorred in the variable names (it's 'correct' in the line above, for "backgroundColor"). Also, if "colours" is not modified anywhere, it should be final.

Reply:
good catch completely missed that. Done